arxlive_spotlight_annotation/src/node_modules/dbpedia/spotlight.mjs

```diff
@@ -26,21 +65,144 @@ export const annotate = async (
+ * @property {number} percentageOfSecondRank - the relative difference in topic score between the first and the second ranked resource.
+ */
+
+/**
+ * 
+ * @typedef ParsedAnnotation
+ * @type {Object}
+ * @property {ReducedResource[]} results - an array of simplified results.
+ * @property {number} confidence - the confidence at which these results were annotated.
+ */
+
+/**
+ * @function parseAnnotationResults
+ * @description this function takes an array of Annotation objects and simplifies them.
+ * @param {SpotlightAnnotation} spotlightAnnotation - an object returned by {@link annotate}
+ * @returns {ParsedAnnotation} parsed Annotation. 
+ */
+export const parseAnnotationResults = (spotlightAnnotation) => {
+	const resultSubset = spotlightAnnotation.Resources
+		? spotlightAnnotation.Resources.map((result) => {
 				return {
-					'@URI': result['@URI'],
-					'@surfaceForm': result['@surfaceForm'],
-					'@similarityScore': result['@similarityScore'],
-					'@percentageOfSecondRank':
-						result['@percentageOfSecondRank'],
+					URI: result['@URI'],
+					surfaceForm: result['@surfaceForm'],
+					similarityScore: result['@similarityScore'],
+					percentageOfSecondRank: result['@percentageOfSecondRank'],
 				};
 		  })
 		: [];
 
 	return {
-		confidence: results['@confidence'],
+		confidence: spotlightAnnotation['@confidence'],
 		results: resultSubset,
 	};
 };
+
+/**
+ * The final form of resource, this is the same as {@link ReducedResource}, however
+ * the confidence property has been added to the objects values.
+ * All properties below are the exact same.
+ * @typedef DBpediaEntity
+ * @type {Object}
+ * @property {string} URI - the Unique Resource Identifier for this resource.
+ * @property {string} surfaceForm - the original string used to produce this resource.
+ * @property {number} similarityScore - cosine similarity between the context vectors and the context surrounding the surface form.
+ * @property {number} percentageOfSecondRank - the relative difference in topic score between the first and the second ranked resource.
+ * @property {number} confidence - the highest confidence at which this resource could be found. This means that all annotations performed at a lower confidence than the one given here will also produce this annotation.
+ */
+
+/**
+ * @function reduceAnnotationResults 
+ * @description Maps {@link ParsedAnnotation} objects using their confidedence.
+ * @param {Object.<string, ParsedAnnotation>} parsedAnnotationByConfidence - an object where {@link ReducedResource} objects are mapped by the condience with which they were produced.
+ * @returns {DBpediaEntity[]} - a list of annotated entities.
+ */
+export const reduceAnnotationResults = (parsedAnnotationByConfidence) => {
+	const seen = new Set();
+	const confidenceKeys = [
+		'1.0',
+		'0.9',
+		'0.8',
+		'0.7',
+		'0.6',
+		'0.5',
+		'0.4',
+		'0.3',
+		'0.2',
+		'0.1',
+		'0.0',
+	];
```

Comment:
Can we reuse or generate these from the config?

Reply:
I think I hard coded these because they were string literals, and wanted to ensure that no rounding happened if casting a float to a string... Thoughts?

Reply:
I think `n.toString()` would work:

```
> n = 0.1
0.1
> n.toString()
'0.1'
> n = 0.9
0.9
> n.toString()
'0.9'
```